### ISSUE TITLE
How to load KITTII Dataset in pointcnn

### ISSUE THREAD

**Issue by tashfin-msu** [rusty1s/pytorch_geometric#2949]
I am trying to use pointCNN for Kittii Dataset.
All of the networks are using get_dataset function to download the modelnet40 dataset. I did not even see the downloaded modelnet in the PC but the code is running fine.
I need to know where to modify the code to let the network run for custom KITTII dataset which is already downloaded in another folder.

Thanks
Ahmed

**Comment by rusty1s** [comment]
I sadly have no experience with KITTI. However, you need to process the dataset in order to obtain point clouds in PyTorch format, *e.g.*, a `pos` matrix of shape `[num_points, 3]`, which you can then input into `PointCNN`.

**Comment by tashfin-msu** [comment]
Hi @rusty1s ,
I could use the Dataloader to load the data perfectly, but now the problem is with the xconv layer when I am trying to run the network, I am getting the following error. 

` deg = x.new_zeros(batch_size, dtype=torch.long)
        deg.scatter_add_(0, batch_x, torch.ones_like(batch_x))
        ~~~~~~~~~~~~~~~~ <--- HERE

        ptr_x = deg.new_zeros(batch_size + 1)
RuntimeError: scatter_(): Expected dtype int64 for index.`

I used both but still getting the same error.  ` y = torch.tensor(1, dtype=torch.long) and y = torch.tensor(1, dtype= torch.int64)`

**Comment by rusty1s** [comment]
Can you identify the reason why `batch_x` is not of type `torch.int64`?

**Comment by tashfin-msu** [comment]
Yes. I added dtype = torch.int64 for batch_x and the above mentioned error is gone, but now I am getting another error:

`Error: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "/home/xxxx/anaconda3/envs/pygeo/lib/python3.8/site-packages/torch_cluster/knn.py", line 55, in knn_graph
    ptr_x: Optional[torch.Tensor] = None
    if batch_x is not None:
        assert x.size(0) == batch_x.numel()
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        batch_size = int(batch_x.max()) + 1
RuntimeError: AssertionError: `

**Comment by rusty1s** [comment]
It looks like you deleted your previous message. Is this issue resolved?

**Comment by tashfin-msu** [comment]
The problem is not solved yet. I changed the batch_x data type as you mentioned and now I am getting a "Segmentation fault (core dumped)" error, which is happening just before training the model.

**Comment by rusty1s** [comment]
Which operation produces the Segmentation fault?
